FuelLabs/fuel-indexer
# chore: improve UX of `forc index` plugin commands

Closes #1135.

### Description

- Removes redundant options from various `forc index` commands
- Fixes a panic in `forc index postgres create --help`
- Removes `forc index init` command
- Updates CI
- Updates documentation
- Updates welcome message in `forc index new`

### Testing steps
1. Create an indexer with `forc index new` and ensure that all given values are respected:
-- `forc index new [path]` 
-- `forc index new [path] --name [name] --namespace [namespace] --native`
3. Run `forc index check` to ensure that components are still detected.
